### ISSUE TITLE
osx: ./etc/create_bridge.sh can't use the ip command

### ISSUE THREAD

**Issue by paul-nelson-baker** [hioa-cs/IncludeOS#896]
The create_bridge.sh script [uses the command ip](https://github.com/hioa-cs/IncludeOS/blob/1d2c42b3b1d338b237fce28e775f53885702b4ca/etc/create_bridge.sh#L25). This does not work on OSX, as there is no direct equivalent at the time.

The closest thing is `brew install iproute2mac`, however, it does not have the same interface as the linux `ip` command. Attempting to use it in place dumps out correct usage instructions.

```
$ ./etc/create_bridge.sh
>>> Network bridge already created
Usage: ip [ OPTIONS ] OBJECT { COMMAND | help }
       ip -V
where  OBJECT := { link | addr | route | neigh }
       OPTIONS := { -4 | -6 }

iproute2mac
Homepage: https://github.com/brona/iproute2mac
This is CLI wrapper for basic network utilities on Mac OS X inspired with iproute2 on Linux systems.
Provided functionality is limited and command output is not fully compatible with iproute2.
For advanced usage use netstat, ifconfig, ndp, arp, route and networksetup directly.
>>> Configuring network bridge (requires sudo):
ifconfig: interface include0 does not exist
```

**Comment by alfred-bratterud** [comment]
Hi, you're right -   I believe@mnordsletten was able to make qemu + networking to work on mac. In the meantime using virtualbox is one alternative we know works, where networking at least is easier to set up. (You can use ./etc/vboxrun.sh <your image> for this or the virtualbox GUI)

**Comment by TheWookie** [comment]
Thanks for the tip! To fellow mac users, this is what he refers to in regard to running examples:
`make && ../../etc/vboxrun.sh "$(make servicefile)"`

**Comment by AndreasAakesson** [comment]
Issue #619 has some information about the equivalent commands to create a bridge on mac.

**Comment by gurka** [comment]
`ip` is just used to check if the bridge is already configured, so that it
can avoid an unnecessary `sudo`. If the `ip` command is missing the script
should always configure the bridge (that's at least what I thought).

What's happening now when you run the script? Does it fail to create and
configure the bridge completely?

Den 21 okt. 2016 7:52 fm skrev "Alfred Bratterud" <notifications@github.com
>:

Hi, you're right - I believe@mnordsletten was able to make qemu +
networking to work on mac. In the meantime using virtualbox is one
alternative we know works, where networking at least is easier to set up.
(You can use ./etc/vboxrun.sh for this or the virtualbox GUI)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/hioa-cs/IncludeOS/issues/896#issuecomment-255299720>,
or mute the thread

**Comment by mnordsletten** [comment]
This has been fixed in #923. This issue can be closed @fwsGonzo
